app/src/main/java/org/mozilla/rocket/home/pinsite/PinSiteManager.kt

```diff
@@ -0,0 +1,265 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.rocket.home.pinsite
+
+import android.content.Context
+import android.preference.PreferenceManager
+import mozilla.components.support.base.log.Log
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import org.mozilla.focus.BuildConfig
+import org.mozilla.focus.R
+import org.mozilla.focus.history.model.Site
+import org.mozilla.focus.utils.TopSitesUtils
+
+fun getPinSiteManager(context: Context): PinSiteManager {
```

Comment:
could you add a comment describing why you choose to create PinSiteManager freely instead of a singleton? I think it'll make it clear for other people to know the pros and cons, e.g: easier testing, it didn't cache state so you can create it freely.